### ISSUE TITLE
Too many levels of symbolic links error on starting container

### ISSUE THREAD

**Issue by perpen** [docker/docker#15679]
I have been successfully running docker on my Chromebook for a few months.
However since a recent update of Chrome OS I am unable to start containers, due to an ELOOP error when docker tries bind-mounting `/proc/self/ns/net`.
A Chrome OS update pushes a new kernel, and new root fs. However I am still using the *same custom kernel and modules* as before the update, leading me to think the problem is due to a change of the root filesystem - but I can't understand what could be impacting docker in such a way. If something else has changed in my environment, I cannot imagine what it could be.
Of course this is not exactly a standard setup, but I would appreciate any pointers.

### docker version
I use the docker binary provided by the latest archlinux package.
```
Client:
 Version:      1.8.1
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   d12ea79
 Built:        Sat Aug 15 17:29:10 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.8.1
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   d12ea79
 Built:        Sat Aug 15 17:29:10 UTC 2015
 OS/Arch:      linux/amd64
```

### docker info
```
Containers: 0
Images: 3
Storage Driver: devicemapper
 Pool Name: docker-254:1-8179-pool
 Pool Blocksize: 65.54 kB
 Backing Filesystem: extfs
 Data file: /dev/loop1
 Metadata file: /dev/loop2
 Data Space Used: 1.825 GB
 Data Space Total: 107.4 GB
 Data Space Available: 15.31 GB
 Metadata Space Used: 1.556 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.146 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.102 (2015-07-07)
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.14.0
Operating System: <unknown>
CPUs: 4
Total Memory: 15.59 GiB
Name: localhost
ID: E7D7:5RGA:JU6A:GWNZ:ACYA:VNQF:DOM6:AF4X:KGRY:RXRG:T2BT:OPGY
Debug mode (server): true
File Descriptors: 11
Goroutines: 19
System Time: 2015-08-18T23:00:37.68208035+01:00
EventsListeners: 0
Init SHA1: 3d05b45242f07d58ebcc51dc1a354a024048d0c7
Init Path: /usr/local/tmp/docker/dockerinit
Docker Root Dir: /var/lib/docker
Username: columnated
Registry: https://index.docker.io/v1/
WARNING: No memory limit support
WARNING: No swap limit support
```
### uname -a
`Linux arch 3.14.0 #1 SMP Tue Aug 18 22:43:56 BST 2015 x86_64 GNU/Linux`
### Environment details (AWS, VirtualBox, physical, etc.):
I run docker on a Chromebook Pixel LS, with a custom kernel.
The output of `./contrib/check-config.sh` is: https://gist.github.com/perpen/6fe3de129df0716ab44e#file-output-of-check-config-sh
The full kernel config: https://gist.github.com/perpen/6fe3de129df0716ab44e#file-kernel-config

### How reproducible:
Always

### Steps to Reproduce:
```
$ /usr/local/tmp/docker-plain/docker -H unix:///tmp/docker.sock run -it --rm busybox
WARNING: Your kernel does not support memory swappiness capabilities, memory swappiness discarded.
Error response from daemon: Cannot start container 52941566207d0ce9b21b5820bd6f3c1c14fdf5334965e63a2254bf3e17ad59df: failed sandbox add: failed to create new sandbox: namespace creation reexec command failed: exit status 1
```
### Additional info
The docker log output shows this:
`level=fatal msg="too many levels of symbolic links" `
Full docker log: https://gist.github.com/perpen/6fe3de129df0716ab44e#file-docker-log-output

Running strace against docker shows this error:
`[pid  5518] mount("/proc/self/ns/net", "/var/run/docker/netns/3d21a7dd2745", 0xc20809b8e0, MS_BIND, NULL) = -1 ELOOP (Too many levels of symbolic links)`

Running manually the command corresponding to the syscall above, ie `mount --bind /proc/self/ns/net /var/run/docker/netns/3d21a7dd2745` does not result in an error.

The same error occurred with my previous version of docker, 1.7.

**Comment by apporc** [comment]
Do you still have this problem now, or find a way to solve this?
I am using latest chromiumos, with kernel 3.18 and docker 1.7.1, same error.

**Comment by perpen** [comment]
@apporc still the same. I switched to beta channel to see if it would make a difference, it didn't.
Did you have docker running with a previous version of Chromium OS?

**Comment by apporc** [comment]
@perpen no, in fact i am porting docker to chromium os on my local branch. the code is from current master branch of upstream. After all i have done, the dev image built for amd64 with docker builtin, can autostart. "docker info/version" all ok, but when i was trying to start a container, the error happened. Do you know from which version on, the error begin to show?

**Comment by perpen** [comment]
@apporc it must have happened with stable release 44.0.2403.155 (Platform version: 7077.123.0), released August 11th. 
I don't know how you relate these versions to Chromium OS tags, but would you be able to apply your changes to the previous stable release, 44.0.2403.125 (Platform version: 7077.111.0) and see if it works?

**Comment by perpen** [comment]
I managed to find the problem: the Chrome OS kernel uses a custom lsm (linux security model, was new to me) which implements a check which refuses to mount a symlink, and returns `-ELOOP` if this is attempted.
Hence the failure when mounting `/proc/self/ns/net` into `/var/run/docker/netns/<blah>`.
I disabled this check by editing `chromiumos_security_sb_mount()` in `security/chromiumos/lsm.c` and with this new kernel I got docker back.
I'll mention it to the Chromium guys, it is disappointing that a custom kernel is still necessary to run docker properly.
